src/components/generic/Tooltip/Tooltip.jsx

```diff
@@ -0,0 +1,76 @@
+import React, {useEffect, useState} from 'react';
+import PropTypes from 'prop-types';
+import webexComponentClasses from '../../helpers';
+import Portal from '../Portal/Portal';
+
+/**
+ * Displays tooltip
+ *
+ * @param {object} props  Data passed to the component
+ * @param {React.ReactNode[]} props.children  List of children
+ * @param {string} [props.className]  Additional className for the component
+ * @param {string} props.style  Inline style for the component
+ * @param {object} props.target  Ref for the target element
+ * @returns {object} JSX of the element
+ */
+const Tooltip = ({
+  children,
+  className,
+  style,
+  target,
+}) => {
+  const cssClasses = webexComponentClasses('tooltip-container', className);
+  const [tooltipVisible, setTooltipVisible] = useState(false);
+  const rect = target.getBoundingClientRect();
+  const coords = {
+    left: rect.x + rect.width / 2, // add half the width of the button for centering
+    top: rect.y + window.scrollY, // add scrollY offset, as soon as getBountingClientRect takes on screen coords
+  };
+  const handleMouseEnter = () => {
```

Comment:
you can create these functions inside the `useEffect` block, so that they are not created when not needed